AnkiDroid/src/main/java/com/ichi2/anki/Preferences.java

```diff
@@ -842,11 +843,13 @@ private String replaceStringIfNumeric(String str, String value) {
 
     private void initializeLanguageDialog(PreferenceScreen screen) {
         ListPreference languageSelection = (ListPreference) screen.findPreference(LANGUAGE);
+        Locale currentAppLocale = LanguageUtil.getLocale(AnkiDroidApp.getSharedPrefs(AnkiDroidApp.getInstance())
+                .getString(Preferences.LANGUAGE, ""));
         if (languageSelection != null) {
             Map<String, String> items = new TreeMap<>();
             for (String localeCode : LanguageUtil.APP_LANGUAGES) {
                 Locale loc = LanguageUtil.getLocale(localeCode);
-                items.put(loc.getDisplayName(), loc.toString());
+                items.put(loc.getDisplayName(currentAppLocale), loc.toString());
```

Comment:
@david-allison-1 separately there is #6252 - seems like we should just put it in here. I understand we're working on sorting that one, I wonder (since the whole area was broken anyway) if we couldn't at least get this functionality in so language switching works at all and then worry about sorting it after? Having this not be in the locale's language means users may become trapped in an accidental but not-decipherable language though, so I'd vote to put loc in here personally

```suggestion
                items.put(loc.getDisplayName(loc), loc.toString());
```

Reply:
I already answered @david-allison-1 in this regard. I've seen that PR and I think I know a simple solution for the outstanding sorting part of the language list. I just wanted to finish this PR first before taking on more tasks.

Reply:
Okay great - this is going in then, thanks so much for fixing this, it's a big item for a lot of people